Src/AutoFixture.xUnit.net.UnitTest/Scenario.cs

```diff
@@ -141,5 +141,85 @@ public void FavorArraysCausesArrayConstructorToBeInjectedWithFrozenItems([Frozen
         {
             Assert.Same(p1, p2);
         }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterShouldAssignSameInstanceToSecondParameter(
+            [Freeze]string p1,
+            string p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByExactTypeShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ExactType)]ConcreteType p1,
+            ConcreteType p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByDirectBaseTypeShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.DirectBaseType)]ConcreteType p1,
+            AbstractType p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByInterfaceShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ImplementedInterfaces)]NoopInterfaceImplementer p1,
+            IInterface p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByParameterNameShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ParameterName, TargetName = "parameter")]string p1,

```

Comment:
Certainly we should document how this work :)

While I admit that this approach isn't refactoring-safe, is the one that involves a `TargetName` property? That's a string as well.

Granted, some sophisticated refactoring tools may be able to detect that string as well, but I wouldn't bet on it.

In any case, since it sounds like you may agree that this is the direction to take, do consider where to put the main implementation of these less strict matching rules. Could they go into the AutoFixture kernel, in the new Specifications you recently added?